airbyte-integrations/connectors/destination-iceberg-v2/src/test-integration/kotlin/io/airbyte/integrations/destination/iceberg/v2/IcebergV2WriteTest.kt

```diff
@@ -0,0 +1,147 @@
+        preserveUndeclaredFields = false,
+        commitDataIncrementally = false,
+        supportFileTransfer = false,
+        allTypesBehavior = StronglyTyped(),
+    ) {
+    companion object {
+        @JvmStatic
+        @BeforeAll
+        fun setup() {
+            NessieTestContainers.start()
+        }
+    }
+}
+
+class IcebergNessieMinioWriteTest : IcebergV2WriteTest(getConfig()) {
+    @Test
+    @Disabled(
+        "Expected because we seem to be mapping timestamps to long when we should be mapping them to an OffsetDateTime"
+    )
+    override fun testBasicTypes() {
+        super.testBasicTypes()
+    }
+
+    @Test
+    @Disabled(
+        "Expected because we seem to be mapping timestamps to long when we should be mapping them to an OffsetDateTime"
+    )
+    override fun testInterruptedTruncateWithPriorData() {
+        super.testInterruptedTruncateWithPriorData()
+    }
+
+    @Test
+    @Disabled("This is currently hanging forever and we should look into why")
+    override fun testInterruptedTruncateWithoutPriorData() {
+        super.testInterruptedTruncateWithoutPriorData()
+    }
+
+    @Test
+    @Disabled
+    override fun testContainerTypes() {
+        super.testContainerTypes()
+    }
+
+    @Test
+    @Disabled(
+        "Expected because we seem to be mapping timestamps to long when we should be mapping them to an OffsetDateTime"
+    )
+    override fun resumeAfterCancelledTruncate() {
+        super.resumeAfterCancelledTruncate()
+    }
+
+    @Test
+    @Disabled("This is expected")
+    override fun testAppendSchemaEvolution() {
+        super.testAppendSchemaEvolution()
+    }
+
+    @Test
+    @Disabled
+    override fun testUnions() {
+        super.testUnions()
+    }
+
+    companion object {
+        private fun getToken(): String {
+            val client = OkHttpClient()
+            val objectMapper = ObjectMapper()
+
+            val credentials = "client1:s3cr3t"
+            val encodedCredentials = Base64.getEncoder().encodeToString(credentials.toByteArray())
+
+            val formBody =
+                FormBody.Builder()
+                    .add("grant_type", "client_credentials")
+                    .add("scope", "profile")
+                    .build()
+
+            val request =
+                Request.Builder()
+                    .url("http://127.0.0.1:8080/realms/iceberg/protocol/openid-connect/token")
+                    .post(formBody)
+                    .header("Content-Type", "application/x-www-form-urlencoded")
+                    .header("Authorization", "Basic $encodedCredentials")
+                    .build()
+
+            val response = client.newCall(request).execute()
+            val jsonResponse = response.body?.string() ?: throw RuntimeException("Empty response")
+            val jsonNode: JsonNode = objectMapper.readTree(jsonResponse)
+            return jsonNode.get("access_token").asText()
+        }
+
+        fun getConfig(): String {
```

Comment:
(nonblocking) at some point we should probably:
* move this to its own file
* make the base CheckIntegrationTest accept file contents instead of file paths
* add a real nessie config to IcebergV2CheckTest